File: src/test/java/com/travel/domain/travel/service/TravelPlanServiceTest.java
package com.travel.domain.travel.service;

import com.travel.domain.travel.domain.plan.TravelPlan;
import com.travel.domain.travel.domain.plan.TravelPlanRepository;
import com.travel.domain.travel.dto.plan.TravelPlanSaveDto;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import java.util.ArrayList;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
public class TravelPlanServiceTest {
    @Autowired
    TravelService travelService;

    @Autowired
    TravelPlanService travelPlanService;

    @Autowired
    TravelPlanRepository travelPlanRepository;

    @DisplayName("[단위테스트] 여행계획 등록")
    @Test
    public void save() {
        //given
        travelPlanService.save(TravelPlanSaveDto.builder()
                .travelId(1)
                .travelDate("20221229")
                .place("동성로")
                .build());

        //when
        List<TravelPlan> list = travelPlanRepository.findAll();

        //then
        assertThat(list.get(0).getPlace()).isEqualTo("동성로");
    }

    @DisplayName("[단위테스트] 여행계획 일괄 등록(Update 기능)")
    @Test
    public void saveAll() {
        List<TravelPlanSaveDto> planSaveDtoList = new ArrayList<>();
        planSaveDtoList.add(TravelPlanSaveDto.builder().travelId(1).travelDate("20230101").place("대학로2").orderNo(1).build());
        planSaveDtoList.add(TravelPlanSaveDto.builder().travelId(1).travelDate("20230101").place("대학로").orderNo(2).build());
        planSaveDtoList.add(TravelPlanSaveDto.builder().travelId(1).travelDate("20230101").place("계명대").orderNo(3).build());
        planSaveDtoList.add(TravelPlanSaveDto.builder().travelId(1).travelDate("20230101").place("서문시장").orderNo(4).build());
        planSaveDtoList.add(TravelPlanSaveDto.builder().travelId(1).travelDate("20230101").place("경북대").orderNo(5).build());
        //given
        travelPlanService.saveAll(planSaveDtoList);

        //when
        List<TravelPlan> list = travelPlanRepository.findAll();

        //then
        assertThat(list.get(0).getPlace()).isEqualTo("대학로2");
    }
}

File: src/test/kotlin/com/travel/domain/travel/TravelPlanServiceTest.kt
package com.travel.domain.travel

import com.travel.domain.travel.domain.plan.TravelPlanRepository
import com.travel.domain.travel.dto.plan.TravelPlanSaveDto
import com.travel.domain.travel.service.TravelPlanService
import com.travel.domain.travel.service.TravelService
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
class TravelPlanServiceTest @Autowired constructor(
    private var travelService: TravelService,
    private var travelPlanService: TravelPlanService,
    private var travelPlanRepository: TravelPlanRepository
) {
    @DisplayName("[단위테스트] 여행계획 등록")
    @Test
    fun save() {
        //given
        travelPlanService.save(TravelPlanSaveDto(
                travelId = 1L,
                travelDate = "20221229",
                place = "동성로"
        ))

        //when
        val list = travelPlanRepository!!.findAll()

        //then
        Assertions.assertThat(list[0].place).isEqualTo("동성로")
    }

    @DisplayName("[단위테스트] 여행계획 일괄 등록(Update 기능)")
    @Test
    fun saveAll() {
        val planSaveDtoList: MutableList<TravelPlanSaveDto> = ArrayList()
        planSaveDtoList.add(TravelPlanSaveDto(travelId = 1, travelDate = "20230101", place = "대학로2", orderNo = 1))
        planSaveDtoList.add(TravelPlanSaveDto(travelId = 1, travelDate = "20230101", place = "대학로", orderNo = 2))
        planSaveDtoList.add(TravelPlanSaveDto(travelId = 1, travelDate = "20230101", place = "계명대", orderNo = 3))
        planSaveDtoList.add(TravelPlanSaveDto(travelId = 1, travelDate = "20230101", place = "서문시장", orderNo = 4))
        planSaveDtoList.add(TravelPlanSaveDto(travelId = 1, travelDate = "20230101", place = "경북대", orderNo = 5))

        //given
        travelPlanService!!.saveAll(planSaveDtoList)

        //when
        val list = travelPlanRepository!!.findAll()

        //then
        Assertions.assertThat(list[0].place).isEqualTo("대학로2")
    }
}